modules/@apostrophecms/asset/index.js

```diff
@@ -288,6 +290,24 @@ module.exports = {
               for (const dir of directories[name]) {
                 const srcDir = `${dir}/${source}`;
                 if (fs.existsSync(srcDir)) {
+                  if (
+                    // is pnpm installation
+                    self.apos.isPnpm &&
+                    // is npm module
+                    npmNames[name] &&
+                    // dir.startsWith(path.join(self.apos.npmRootDir, 'node_modules')) &&
+                    // is not apos core module
+                    !dir.startsWith(path.join(self.apos.npmRootDir, 'node_modules/apostrophe')) &&
+                    // is not `my-*` module
+                    dir.endsWith(name) &&
+                    // is not a bundle sub-module
```

Comment:
There is no fixed rule that bundles always have this naming pattern. It's just our convention.

However since we always follow it we could document it as a requirement for pnpm compatibility.